lib/Sema/SemaDecl.cpp

```diff
@@ -10868,6 +10868,30 @@ void Sema::ActOnFinishKNRParamDeclarations(Scope *S, Declarator &D,
   }
 }
 
+/// ActOnBoundsExpr: attach a bounds expression to a parameter declaration.
+void Sema::ActOnBoundsExpr(Decl *D, Expr *BoundsExpr) {
+  if (!D || !BoundsExpr)
+    return;
+
+  VarDecl *Var = cast<VarDecl>(D);
```

Comment:
Does this throw or otherwise fail in a noticeable way if the Decl isn't a VarDecl?  It would be nice to know as soon as that happens rather than end up corrupting memory when we call setBoundsExpr below.